File: android/app/src/main/java/com/harshbshah/fer/data/model/ActiveSessionSnapshot.kt
package com.harshbshah.fer.data.model

import com.google.firebase.firestore.PropertyName
import java.util.Date

/**
 * The cross-device live-workout document, shared with iOS via Firestore at
 * users/{uid}/activeSession/current — mirrors Fer/Models/ActiveSessionSnapshot.swift
 * field-for-field. Whichever device is active writes the complete state here;
 * `updatedAt` is what lets each side tell its own echo apart from a genuine
 * edit made on the other device.
 */
data class ActiveSessionSnapshot @JvmOverloads constructor(
    var routineName: String = "",
    var exercises: List<LoggedExercise> = emptyList(),
    var startedAt: Date = Date(),

    @get:PropertyName("isResting") @set:PropertyName("isResting")
    var isResting: Boolean = false,

    var restEndDate: Date? = null,
    var restSecondsByExerciseId: Map<String, Int> = emptyMap(),
    var weightUnitRaw: String = "lb",
    var updatedAt: Date = Date()
)

File: Fer/Models/ActiveSessionSnapshot.swift
//
//  ActiveSessionSnapshot.swift
//  Fer
//
//  The cross-device live-workout document, shared between iOS and Android
//  via Firestore at users/{uid}/activeSession/current — the cloud analog of
//  WorkoutMirrorSnapshot (which only reaches the paired Watch over
//  WatchConnectivity). Whichever device is active writes the complete state
//  here; `updatedAt` is what lets each side tell its own echo apart from a
//  genuine edit from the other device (see WorkoutSessionViewModel's
//  lastPushedUpdatedAt / applyRemoteSnapshot).
//

import Foundation

struct ActiveSessionSnapshot: Codable, Equatable {
    var routineName: String
    var exercises: [LoggedExercise]
    var startedAt: Date
    var isResting: Bool
    var restEndDate: Date?
    var restSecondsByExerciseId: [String: Int]
    var weightUnitRaw: String
    var updatedAt: Date
}
